tools/format/CMakeLists.txt

```diff
@@ -4,76 +4,68 @@
 cmake_minimum_required(VERSION 3.29.0)
 project(msvc_standard_libraries_format NONE)
 
-set(did_search OFF)
-
-if(NOT DEFINED CLANG_FORMAT)
-    message(STATUS "Searching for VS clang-format")
-    set(did_search ON)
-endif()
-
-if(PROJECT_IS_TOP_LEVEL)
-    set(message_level FATAL_ERROR)
-else()
-    set(message_level WARNING)
-endif()
-
 find_program(CLANG_FORMAT
     NAMES clang-format
-    PATHS "C:/Program Files/Microsoft Visual Studio/2022/Preview/VC/Tools/Llvm/bin"
+    HINTS "$ENV{VCINSTALLDIR}/Tools/Llvm/$ENV{VSCMD_ARG_HOST_ARCH}/bin"

```

Comment:
`VSCMD_ARG_HOST_ARCH` is probably wrong.

I have only `C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\Llvm\x64\bin` and `C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\Llvm\ARM64\bin`

but `%VSCMD_ARG_HOST_ARCH%` is `x86` if I open `x86 Native tools` and I want to build `x86` library.

![изображение](https://github.com/user-attachments/assets/82190938-880a-4f81-8987-7823988f562e)

And because ARM machines can run x64 binary I think we can change `$ENV{VSCMD_ARG_HOST_ARCH}` to `x64`. What do you think?

Reply:
Ugh, yes, I forgot that x86 LLVM isn't in an `x86` directory as is the case for `x64` and `arm64`. I agree we should simply default to `x64` for now. We can devise something clever if anyone ever complains.